magento/pwa-studio
# Change style-loader to something SSR-ready

## Description

In the course of removing (or guarding) `window` and other browser-only global references from the codebase, it became clear that a few references existed in dependency code. One such dependency is `style-loader`, a standard Webpack loader that inserts CSS into `<style>` elements. The maintainers of `style-loader` appear to be aware of the issue, [but have closed it as `won't-fix`](https://github.com/webpack-contrib/style-loader/issues/439), so we need to replace it.

The closest server-compatible loader I can find is [`isomorphic-style-loader`](https://www.npmjs.com/package/isomorphic-style-loader); while it has some minor flaws (hard dependency on `react`) and doesn't appear to be actively maintained anymore, it is MIT licensed, so we should be able to incorporate it with proper attribution. I've tried to do so in this PR.

The largest change proposed here is a switch from `mergeClasses` to a new hook (`useStyle`) with context, which allows style accumulation and insertion to work differently between browser and server.

## Related Issue

## Acceptance

### Verification Stakeholders
- @dpatil-magento 

### Specification

### Verification Steps
1. Verify the app builds without errors (`yarn watch:venia`).
2. Verify there are no console errors.
3. Verify the site doesn't render differently from `develop`.

## Screenshots / Screen Captures (if appropriate)

## Checklist

* I have added tests to cover my changes, if necessary.
* I have added translations for new strings, if necessary.
* I have updated the documentation accordingly, if necessary.
